sklearn/utils/tests/test_pprint.py

```diff
@@ -346,6 +346,14 @@ def test_deeply_nested(print_changed_only_false):
     assert rfe.__repr__() == expected
 
 
+def test_depth(print_changed_only_false):
+    pp = _EstimatorPrettyPrinter(depth=1)
+
+    rfe = RFE(RFE(RFE(RFE(RFE(LogisticRegression())))))
+    expected = "RFE(estimator=RFE(...), n_features_to_select=None, step=1, verbose=0)"
+    assert pp.pformat(rfe) == expected
+
```

Comment:
could you please test here by setting the config_context for `changed_only` to True and False and test both cases explicitly?

Reply:
Done!